doc.go

```diff
@@ -412,8 +414,74 @@ data types. The columns are:
     [5] You cannot directly Scan() into the alternative data types via snowflakeRows.Scan(), but can convert to
     those data types by using .Float32()/.String()/.Float64() methods. For an example, see below.
 
+    [6] Arrays and objects can be either semistructured or structured, see more info in section below.
+
 Note: SQL NULL values are converted to Golang nil values, and vice-versa.
 
+## Semistructured and structured types
+
+Snowflake supports two flavours of "structured data" - semistructured and structured.
+Semistructured types are variants, objects and arrays without schema.
+These types are represented by strings and should be interpreted on the client side.
```

Comment:
I think this makes it more clear `When data is fetched, it's represented as strings and the client is responsible for its interpretation.`

Reply:
Done.